projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -566,22 +563,28 @@ function wpcom_get_published_non_headstart_posts_count() {
  *
  * @return void
  */
-function wpcom_track_write_3_posts_task() {
-	$published_non_headstart_posts = wpcom_get_published_non_headstart_posts_count();
+function wpcom_launchpad_track_write_3_posts_task() {
+	// If the task is already completed, don't do anything.
+	if ( wpcom_is_task_option_completed( array( 'id' => 'write_3_posts' ) ) ) {
+		return;
+	}
+
+	$published_non_headstart_posts = wpcom_launchpad_get_published_non_headstart_posts_count();
 
 	if ( $published_non_headstart_posts >= 3 ) {
-		wpcom_mark_launchpad_task_complete_if_active( 'write_3_posts' );
+		wpcom_mark_launchpad_task_complete( 'write_3_posts' );
 	}
 }
+add_action( 'publish_post', 'wpcom_launchpad_track_write_3_posts_task', 10, 3 );
```

Comment:
Do we need the 10 or the 3 here? The callback doesn't take any arguments, and the default priority for actions is 10.
```suggestion
add_action( 'publish_post', 'wpcom_launchpad_track_write_3_posts_task' );
```